### ISSUE TITLE
[Adjust Rule] SIM115 false positive in a complex with statement

### ISSUE THREAD

**Issue by danny0838** [MartinThoma/flake8-simplify#152]
The following case gets a SIM115 error, while it shouldn't.

```python
file = '/path/to/file'
with (open(file, 'rb') if file is not None else nullcontext(sys.stdin)) as fh:
    text = fh.read()
```
